From: Darrick J. Wong <djwong@us.ibm.com>
Date: Wed, 24 Nov 2010 16:40:33 +1100
Subject: [PATCH] md: Call blk_queue_flush() to establish flush/fua support
X-Git: be20e6c67b6c6024c19094d3447f144d118733b0
References: FATE#311692
Patch-Mainline: 2.6.34

Before 2.6.37, the md layer had a mechanism for catching I/Os with the
barrier flag set, and translating the barrier into barriers for all
the underlying devices.  With 2.6.37, I/O barriers have become plain
old flushes, and the md code was updated to reflect this.  However,
one piece was left out -- the md layer does not tell the block layer
that it supports flushes or FUA access at all, which results in md
silently dropping flush requests.

Since the support already seems there, just add this one piece of
bookkeeping.

Signed-off-by: Darrick J. Wong <djwong@us.ibm.com>
Signed-off-by: NeilBrown <neilb@suse.de>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/md.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index febd7dd..f7cf16e 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -4280,6 +4280,8 @@ static int md_alloc(dev_t dev, char *name)
 		error = 0;
 	}
 	mutex_unlock(&mddev->open_mutex);
+
+	blk_queue_flush(mddev->queue, REQ_FLUSH | REQ_FUA);
  abort:
 	mutex_unlock(&disks_mutex);
 	if (!error) {
-- 
1.6.0.2

